src/components/header/breadcrumb/generateBreadcrumbs.ts

```diff
@@ -0,0 +1,19 @@
+import { BreadcrumbItem } from "./Breadcrumb";
+
+export const generateBreadcrumbs = (pathname: string): BreadcrumbItem[] => {
+  const breadcrumbs: BreadcrumbItem[] = [];
+  const pathSegments = pathname.split("/").filter((segment) => segment);
+
+  if (pathSegments.length === 1) {
+    return [{ text: "Projects overview" }];
+  }
+
+  if (pathSegments.length > 1) {
+    breadcrumbs.push({ text: pathSegments[1], isDropdown: true });
```

Comment:
Personally (but it's totally up to you) I never use `push` as this function mutates the original array, meaning if someone reads this code he needs to fully understand the flow to guess the final state.

Prefereably I would use `.map` + `.filter`